### ISSUE TITLE
New Rule: Figma Personal Access Token (@secretlint/secretlint-rule-figma)

### ISSUE THREAD

**Issue by azu** [secretlint/secretlint#1447]
## Description

Add a new rule to detect Figma Personal Access Tokens (PATs).

Figma PATs use the `figd_` prefix, making them reliably detectable with low false positive rate.

## Source

- [Figma Personal Access Tokens documentation](https://help.figma.com/hc/en-us/articles/8085703771159-Manage-personal-access-tokens) — **does not document the `figd_` prefix or token format**
- [Figma REST API Authentication](https://developers.figma.com/docs/rest-api/authentication/) — **does not document the `figd_` prefix or token format**
- [GitHub Secret Scanning Push Protection: supported](https://docs.github.com/en/code-security/secret-scanning/introduction/supported-secret-scanning-patterns)
- The `figd_` prefix is known only through community observation; no authoritative non-official source (gitleaks, TruffleHog) has a documented rule for this prefix.

Note: Figma's token format documentation is entirely absent from public sources. The prefix may change in the future.

## Format

Prefix `figd_` followed by 40–200 alphanumeric, underscore, or hyphen characters (based on community observation; observed samples range from ~45 to ~160 characters).

Regex (with Unicode boundary and trailing boundary, ReDoS-safe):
```
/(?<!\p{L})figd_[A-Za-z0-9_-]{40,200}(?![A-Za-z0-9_-])/gu
```

Note: A legacy Figma PAT format with UUID-like structure (`XXXX-XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX`, no `figd_` prefix) may still be valid but is intentionally excluded because the shape is too generic to detect without excessive false positives.

## Analysis

- `figd_` is not officially documented, so the prefix itself could change in the future. This is the highest-uncertainty rule in this batch.
- Original range 50–350 was too wide. Community-observed samples are in the 45–160 range, so 40–200 is a tighter but still inclusive bound. Lower bound dropped from 50 to 40 to cover observed 45–46 character samples.
- gitleaks and TruffleHog do not provide a rule for this prefix — an additional confidence signal is that GitHub Secret Scanning does support it.
- Added trailing boundary `(?![A-Za-z0-9_-])` to match the token exactly.

## OK (should NOT be detected)

```
# placeholder / too short
figd_abc
```

## NG (should be detected)

```
FIGMA_TOKEN=figd_aBcDeFgHiJkLmNoPqRsT1234567890abcdefghijklmnopqrstuvwxyz
```
